### DISCUSSION TITLE
How can Layouts get Reference/Viewport Dimensions (Bounding Box)

### DISCUSSION THREAD

**Discussion by mbehrisch** [graphology/graphology#508] [Q&A] [answered]
Hi, 

I would like to use the Standard Library's Layouts (`https://graphology.github.io/standard-library/layout.html`). However, I am struggling to tell the layouts in which bounding box they might operate. 

The circle layout uses `(0,0)` as its mid-point leading to 3/4 points being out of the viewport. 
The same applies to Random, Noverlap, ForceAtlas2, etc.

Thanks a lot in advance!
Michael Behrisch

**Comment by Yomguithereal** [comment, marked as answer]
> The circle layout uses (0,0) as its mid-point leading to 3/4 points being out of the viewport.

What viewport are you speaking about? `graphology` does not render anything. Are you drawing yourself on canvas? Are you using a third-party renderer?

Most layouts operate in an unbounded euclidean space whose dimensions range from -infinity to infinity. It's up to you to rescale them to a specific viewport by computing their actual bounding box to rescale them into another coordinate system if you need to. That's exactly what `graphology-canvas` does internally or what third-party renderers like `sigma.js` do.

**Reply by mbehrisch** [reply]
This is perfect. Exactly what I needed. I was unaware of `graphology-canvas` I knew about the sigma demos, but this one helps.

Thanks a lot!!!
